src/Skia/Avalonia.Skia/FramebufferRenderTarget.cs

```diff
@@ -1,82 +1,201 @@
+            };
+
+            return new DrawingContextImpl(createInfo, _preFramebufferCopyHandler, framebuffer);
+        }
+
+        /// <summary>
+        /// Check if two images info are compatible.
+        /// </summary>
+        /// <param name="currentImageInfo">Current.</param>
+        /// <param name="desiredImageInfo">Desired.</param>
+        /// <returns>True, if images are compatible.</returns>
+        private static bool AreImageInfosCompatible(SKImageInfo currentImageInfo, SKImageInfo desiredImageInfo)
+        {
+            return currentImageInfo.Width == desiredImageInfo.Width &&
+                   currentImageInfo.Height == desiredImageInfo.Height &&
+                   currentImageInfo.ColorType == desiredImageInfo.ColorType;
         }
 
-        class PixelFormatShim : IDisposable
+        /// <summary>
+        /// Create Skia surface backed by given framebuffer.
+        /// </summary>
+        /// <param name="desiredImageInfo">Desired image info.</param>
+        /// <param name="framebuffer">Backing framebuffer.</param>
+        private void CreateSurface(SKImageInfo desiredImageInfo, ILockedFramebuffer framebuffer)
         {
-            private readonly SKImageInfo _nfo;
-            private readonly IntPtr _fb;
-            private readonly int _rowBytes;
-            private SKBitmap _bitmap;
+            if (_framebufferSurface != null && AreImageInfosCompatible(_currentImageInfo, desiredImageInfo) && _currentFramebufferAddress == framebuffer.Address)
+            {
+                return;
+            }
+            
+            FreeSurface();
+            
+            _currentFramebufferAddress = framebuffer.Address;
+
+            var surface = SKSurface.Create(desiredImageInfo, _currentFramebufferAddress, framebuffer.RowBytes);
+
+            // If surface cannot be created - try to create a compatibilty shim first
+            if (surface == null)
+            {
+                _conversionShim = new PixelFormatConversionShim(desiredImageInfo, framebuffer.Address);
+                _preFramebufferCopyHandler = _conversionShim.SurfaceCopyHandler;
 
-            public PixelFormatShim(SKImageInfo nfo, IntPtr fb, int rowBytes)
+                surface = _conversionShim.Surface;
+            }
+
+            _framebufferSurface = surface ?? throw new Exception("Unable to create a surface for pixel format " +
+                                                                 framebuffer.Format +
+                                                                 " or pixel format translator");
+            _currentImageInfo = desiredImageInfo;
+        }
+
+        /// <summary>
+        /// Free Skia surface.
+        /// </summary>
+        private void FreeSurface()
+        {
+            var ownsSurface = _conversionShim != null;
+
+            _conversionShim?.Dispose();
+            _conversionShim = null;
+            _preFramebufferCopyHandler = null;
+
+            if (ownsSurface)
```

Comment:
Sorry, but if  declaration(`var ownsSurface = _conversionShim != null;`) will be near use, it will be a little prettier.

Reply:
Agreed. I will make the change.